concrete-core-fixture/src/generation/synthesizing/glwe_ciphertext.rs

```diff
@@ -63,4 +101,223 @@ mod backend_core {
+            let polynomial_size = entity.polynomial_size();
+            let slice = self
+                .core_engine
+                .consume_retrieve_glwe_ciphertext(entity)
+                .unwrap();
+            let ciphertext = self
+                .core_engine
+                .create_glwe_ciphertext(
+                    unsafe { Vec::from_raw_parts(slice.as_mut_ptr(), slice.len(), slice.len()) },
+                    polynomial_size,
+                )
+                .unwrap();
+            ProtoBinaryGlweCiphertext32(ciphertext)
+        }
+
+        fn destroy_glwe_ciphertext_mut_view(&mut self, entity: GlweCiphertextMutView32) {
+            // There should be only one mutable reference to memory, so re-construct the vector so
+            // that it frees memory when it's dropped
+            let slice = self
+                .core_engine
+                .consume_retrieve_glwe_ciphertext(entity)
+                .unwrap();
+            unsafe { Vec::from_raw_parts(slice.as_mut_ptr(), slice.len(), slice.len()) };
+        }
+    }
+
+    impl<'a> SynthesizesGlweCiphertextMutView<Precision64, GlweCiphertextMutView64<'a>> for Maker {
+        fn synthesize_glwe_ciphertext_mut_view(
+            &mut self,
+            prototype: &Self::GlweCiphertextProto,
+        ) -> GlweCiphertextMutView64<'a> {
+            let ciphertext = prototype.0.to_owned();
+            let polynomial_size = ciphertext.polynomial_size();
+            let container = self
+                .core_engine
+                .consume_retrieve_glwe_ciphertext(ciphertext)
+                .unwrap();
+            self.core_engine
+                .create_glwe_ciphertext(container.leak(), polynomial_size)
+                .unwrap()
+        }
+
+        fn unsynthesize_glwe_ciphertext_mut_view(
+            &mut self,
+            entity: GlweCiphertextMutView64,
+        ) -> Self::GlweCiphertextProto {
+            let polynomial_size = entity.polynomial_size();
+            let slice = self
+                .core_engine
+                .consume_retrieve_glwe_ciphertext(entity)
+                .unwrap();
+            let ciphertext = self
+                .core_engine
+                .create_glwe_ciphertext(
+                    unsafe { Vec::from_raw_parts(slice.as_mut_ptr(), slice.len(), slice.len()) },
+                    polynomial_size,
+                )
+                .unwrap();
+            ProtoBinaryGlweCiphertext64(ciphertext)
+        }
+
+        fn destroy_glwe_ciphertext_mut_view(&mut self, entity: GlweCiphertextMutView64) {
+            // There should be only one mutable reference to memory, so re-construct the vector so
+            // that it frees memory when it's dropped
+            let slice = self
+                .core_engine
+                .consume_retrieve_glwe_ciphertext(entity)
+                .unwrap();
+            unsafe { Vec::from_raw_parts(slice.as_mut_ptr(), slice.len(), slice.len()) };
```

Comment:
again

Reply:
See above

Reply:
this won't be a problem with the reworked design thanks to Alex, resolving